students/Kristy_Martini/lesson05/mailroom.py

```diff
@@ -0,0 +1,129 @@
+        for value in newReport.sorted_dict.values():
+            line_str = '{0:21}'.format(value.name) + "$" + '{0:14}'.format(value.total_gift_value) + '{0:12}'.format(value.num_gifts) + " $" + '{0:12}'.format(value.average_gift)
+            print(line_str)
+        print("\n")
+        prompt_user(newReport)
+
+class Donor:
+    """ Donor class stores the info associated with each donor"""
+    def __init__(self, name, gift_value=0, num_gifts=1, total_gift_value=0):
+        self.name = name
+        self.gift_value = gift_value
+        self.num_gifts = num_gifts
+        self.total_gift_value = total_gift_value
+        try:
+            self.average_gift = self.total_gift_value/self.num_gifts
+        except ZeroDivisionError:
+            self.average_gift = 0
+
+    def add_gift(self, amount):
+        """ Add gift amount to a donor already present in database"""
+        self.total_gift_value = self.total_gift_value + amount
+        self.num_gifts += 1
+        self.average_gift = self.total_gift_value/self.num_gifts
+
+def check_name(newReport=None):
+    """ Check the name of the donor before sending a thank you, add to donor list if they do not exist"""
+    if newReport is None:
+        newReport = Report()
+
+    print("To whom would you like to send a thank you?")
+    name_to_thank = input("Please enter the full name of the donor you'd like to thank, or enter 'list' to view a list of current donors: ")
+    if name_to_thank == "quit":
+        prompt_user(newReport)
+    if name_to_thank == "list":
+        newReport.create_report()
+        print("Would you like to thank a donor from this list?")
+        name_to_thank= input("Please enter the full name of the donor you'd like to thank, or enter 'quit' to exit: ")
+        if name_to_thank == "quit":
+            prompt_user(newReport)
+    try: 
+        donor = newReport.donors[name_to_thank]
+    except KeyError:
+        print(name_to_thank, " is a new donor. They will be added to our database.")
+        newDonor = Donor(name_to_thank, 0, 0, 0)
+        newReport.add_donor(newDonor)
+        donor = newReport.donors[newDonor.name]
+    finally:
+        donation_amount = input("Please enter the donation amount for which you want to thank this donor: ")
+        if donation_amount == "quit":
+            prompt_user(newReport)
+        donor.add_gift(int(donation_amount))
+        send_thank_you(donor)
+        prompt_user(newReport)
+
+def send_thank_you(donor):
+    """ Print a thank you to the donor"""
+    output_dir = os.getcwd()
+    output_file = donor.name + ".txt"
+    with open(os.path.join(output_dir, output_file), 'w') as f:
+        f.write("Thank you " + donor.name + " for your charitable gift to our organization.\n We could not operate without the generostiy of donors like yourself.")
+        f.write("Your generous gifts swill allow us to continue to serve our community in the hopes of a better world")
```

Comment:
This should contain the amount given - note the instructions from the prior version:

"Dear Jeff Bezos,

        Thank you for your very kind donation of $877.33.

        It will be put to very good use.

                       Sincerely,
                          -The Team"